### PR TITLE
DM-36947: Forward PYTHONWARNINGS to tests.

### PR DESCRIPTION
**Pull request by TallJimbo**

### CODE REVIEW

File: python/lsst/sconsUtils/tests.py [hunk 1]
```diff
@@ -328,12 +336,15 @@ def runPythonTests(self, pyList):
...
+        # If user has set an envvar to control warnings, defer to them.
+        warn_opt = "-Wd " if "PYTHONWARNINGS" not in os.environ else ""
+
         # Now set up the python testing target
         # We always want to run this with the tests target.
         # We have decided to use pytest caching so that on reruns we only
         # run failed tests.
         lfnfOpt = "none" if 'install' in SCons.Script.COMMAND_LINE_TARGETS else "all"
-        interpreter = f"pytest -Wd --lf --lfnf={lfnfOpt}"
+        interpreter = f"pytest {warn_opt}--lf --lfnf={lfnfOpt}"
```
Comment (TallJimbo) [inline python/lsst/sconsUtils/tests.py:347]: I don't know why we have two versions of this line, but I'd rather not try to remove the duplication on this ticket.

Comment (timj) [reply python/lsst/sconsUtils/tests.py:347]: one is for running the standalone tests and the other is for the "run them all together" tests.

File: python/lsst/sconsUtils/tests.py [hunk 2]
```diff
@@ -235,13 +240,16 @@ def run(self, fileGlob):
         # to the command.
         libpathstr = utils.libraryLoaderEnvironment()
 
+        # If user has set an envvar to control warnings, defer to them.
+        warn_opt = "-Wd " if "PYTHONWARNINGS" not in os.environ else ""
```
Comment (ktlim) [inline python/lsst/sconsUtils/tests.py:244]: I think that `PYTHONWARNINGS` extends (and potentially overrides) anything specified with `-W`, according to my reading of https://docs.python.org/3/library/warnings.html#describing-warning-filters.  So I'm not sure that it's necessary to remove `-Wd`.

Comment (TallJimbo) [reply python/lsst/sconsUtils/tests.py]: Good catch; fixed.  I was led astray by previous versions of this change that suffered from other problems.  I've tested this with my ticket branches for DM-31924 and I can confirm that I see test failures for deprecated interface usage with PYTHONWARNINGS set and these changes removed.

**Pull request merged**